src/schnetpack/interfaces/ensemble_calculator.py

```diff
@@ -0,0 +1,136 @@
+import logging
+import numpy as np
+import torch
+from typing import List, Optional, Dict, Callable, Union
+
+
+class EnsembleAverageStrategy:
+    """
+    base class for Ensemble Average Strategies
+    """
+    def __init__(self):
+        pass
+
+    def correct_dimension(self, num_atoms, inputs:torch.Tensor):
+
+        """
+        Args:
+            inputs:
+                stacked output tensors of predicted property (e.g Energy or Forces)
+            num_atoms:
+                number of atoms in mol. Needed for correct dimension reshaping
+
+        Returns:
+            correct dimension for reshaping the inputs accordingly
+            [num_models, num_mols in batch, num_atoms in mol, property dimension]
+            this way no distinction between single point and batchwise optimization has to be done in Ensemble
+        """
+
+        n_models = inputs.shape[0]
+        batch_size = num_atoms.size()[0]
+        n_atoms = num_atoms.unique()[0].item()
+        property_dim = inputs.shape[-1]
+
+        if len(inputs.shape) == 2:
+            n_atoms = 1
+            property_dim = 1
+
+        return (n_models,batch_size,n_atoms,property_dim)
+
+    def uncertainty_estimation(self, inputs: torch.Tensor, num_atoms):
+        """
+        Args:
+            inputs:
+                stacked output tensors of predicted property (e.g Energy or Forces)
+            num_atoms:
+                number of atoms in mol. Needed for correct dimension reshaping
+
+        Returns:
+            custom uncertainty estimation
+        """
+        raise NotImplementedError
+    
+    def fallback(self, conditions):
+        if conditions.sum() == 0:
+            logging.info(
+                f"All models fail to predict properties with the given filter criteria: {self.filter_criteria.item()}\n"
+                f"Please consider to change the filter criteria or" 
+                f"to lower the applied model drop threshold of currently {self.model_drop_threshold * 100} % "
+                f"Per default now only the first model is considered for the current step"
+            )
+            conditions[0] = True
+        else:
+            pass
+
+        return conditions
+
+
+class SimpleEnsembleAverage(EnsembleAverageStrategy):
```

Comment:
If I understand correctly, this strategy does not just compute mean and std, but also filters out outlier predictions?
Maybe we should also add the most simple average strategy which just computes mean and std. This could then be used as a default in the ensemble calculator.

Reply:
Done